gateway.go

```diff
@@ -317,40 +341,58 @@ func (gw *Gateway) getMatchingAddresses(indexKeySets [][]string) []netip.Addr {
 func (gw *Gateway) Name() string { return thisPlugin }
 
 // A does the A-record lookup in ingress indexer
-func (gw *Gateway) A(name string, results []netip.Addr) (records []dns.RR) {
+func (gw *Gateway) A(name string, results []string) (records []dns.RR) {
```

Comment:
Please keep your hands off of unrelated code to your problem.

Reply:
It is related due to the new map[string][]string that replaces the []netip.Addr due to the TXT records not being up addresses